packages/grafana-runtime/src/services/pluginExtensions/links.ts

```diff
@@ -0,0 +1,46 @@
+import { UrlQueryMap, urlUtil } from '@grafana/data';
+
+import { getPluginsExtensionRegistry } from './registry';
+
+export type PluginLinkOptions = {
+  id: string;
+  queryParams?: UrlQueryMap;
+};
+
+export type PluginLinkResult = {
+  link?: PluginLink;
+  error?: Error;
+};
+
+export type PluginLink = {
+  href: string;
```

Comment:
I think the names of the properties should probably go hand-in-hand with what we have in our schema and [main types](https://github.com/grafana/grafana/pull/61663/files#diff-969e280743cb22049967d8f6cd9b61a7337b5c4b8dd888389bbfa67d8134428dR7) (`href` vs `path`). What do you think?